src/app/githubReleaseNotification.service.js

```diff
@@ -0,0 +1,127 @@
+ */
+
+const semver = require('semver')
+const { Notification, shell } = require('electron')
+const { fetch } = require('undici')
+const packageJson = require('../../package.json')
+
+/**
+ * Request latest release with GitHub REST API and get version from tag name
+ *
+ * @see https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release
+ * @return {Promise<string>} Version tag, for example, v0.2.1
+ */
+async function getLatestReleaseVersion() {
+	try {
+		const response = await fetch('https://api.github.com/repos/nextcloud-releases/talk-desktop/releases/latest', {
+			headers: {
+				Accept: 'application/vnd.github+json',
+				'X-GitHub-Api-Version': '2022-11-28',
+			},
+		})
+		const data = await response.json()
+		if (response.ok) {
+			return data?.tag_name
+		} else {
+			console.error(data)
+		}
+	} catch (e) {
+		console.error(e)
+	}
+}
+
+/**
+ * Show native notification about new version. Opens releases page on click
+ *
+ * @param {string} version - New version
+ */
+function notifyAboutNewVersion(version) {
+	const notification = new Notification({
+		title: '🎉 New version of Nextcloud Talk is available!',
+		body: `Nextcloud Talk ${version} is now available to download from the release page. Click to open the page.`,
+	})
+	notification.on('click', () => {
+		shell.openExternal('https://github.com/nextcloud-releases/talk-desktop/releases/latest')
+	})
+	notification.show()
+}
+
+/**
+ * Check if there is a new Nextcloud Talk
+ *
+ * @param {boolean} [showNotification=false] Show native notification if new version is available
+ * @return {Promise<boolean>} true if new version was found
+ */
+async function checkForNewVersion(showNotification = false) {
+	const latest = await getLatestReleaseVersion()
+	// Something goes wrong. No worries, we will try again later.
+	if (!latest) {
+		return false
+	}
+	if (semver.lte(latest, packageJson.version)) {
+		return false
+	}
+	if (showNotification) {
+		notifyAboutNewVersion(latest)
+	}
+	return true
+}
+
+/** @type {number} setInterval id of the scheduler */
+let schedulerIntervalId
+
+/**
+ * Start scheduler with regular update checks
+ *
+ * @param {number} [intervalInMin=60] Checking interval in minutes
+ * @return {void}
+ */
+function setupReleaseNotificationScheduler(intervalInMin = 60) {
+	// Before run the scheduler - check and stop running scheduler
+	if (schedulerIntervalId !== undefined) {
+		stopReleaseNotificationScheduler()
+	}
+	checkForNewVersion(true)
+	const MS_IN_MIN = 60 * 1000
+	schedulerIntervalId = setInterval(() => {
+		checkForNewVersion(true)
+	}, intervalInMin * MS_IN_MIN)
+}
+
+/**
+ * Stop the scheduler if any
+ *
+ * @return {void}
+ */
+function stopReleaseNotificationScheduler() {
+	if (schedulerIntervalId) {
+		clearInterval(schedulerIntervalId)
+	}
+}
+
+module.exports = {
+	getLatestReleaseVersion,
+	notifyAboutNewVersion,
+	checkForNewVersion,
+	setupReleaseNotificationScheduler,
+	stopReleaseNotificationScheduler,
```

Comment:
![grafik](https://user-images.githubusercontent.com/213943/229724489-5f2e8dc5-c75f-4ed8-9019-9d9fead46411.png)


PS most of them seem to not be used outside, do you export by default? Or does it make sense to not export it?